verilog/formatting/token_annotator.cc

```diff
@@ -93,9 +93,9 @@ static WithReason<int> SpacesRequiredBetween(const PreFormatToken& left,
   }
 
   // Unary operators (context-sensitive)
-  if (IsUnaryPrefixExpressionOperand(left, context)) {
+  if (IsUnaryPrefixExpressionOperand(left, context) &&
+      !IsUnaryOperator(static_cast<yytokentype>(right.TokenEnum()))) {
```

Comment:
Will this preclude `--++x` and `~--x` and `~~x` (as weird as they may seem)?
I actually have a plan to use the lexer itself to pre-compute all illegal 0-space combinations of tokens (internal feature request, not yet published).

I suggest filing a separate issue for this (I'll establish the link) leaving this specific matter to be addressed separately.  It's ok to have a known bug+TODO with fixing spacing around `&`.

Reply:
And I can't think of an example where `-` before `=` is legal.  It is possible to prove as such with a static analysis of the bison-generated automata.  They even provide some summaries of where tokens can appear at the start/end of nonterminals.

Reply:
Thank you for the review!

You are right that some unary combination were broken with my previous patch. I've updated it to check whether the currently processed token can in fact be both binary and unary, if so, and the right token is unary, it is treated as a binary token (this handles the `& ~` case.

Otherwise both tokens are treated as unary, which should resolve the remaining cases: `~--x` and `~~x` work - I don't think `--++x` is legal though.

I also added some integration and unit tests. Please let me know what you think about the updates. Thanks!